src/DependencyInjection/Compiler/TemplateRegistryCompilerPass.php

```diff
@@ -0,0 +1,126 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of the Sonata Project package.
+ *
+ * (c) Thomas Rabaix <thomas.rabaix@sonata-project.org>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sonata\AdminBundle\DependencyInjection\Compiler;
+
+use Sonata\AdminBundle\Admin\AbstractAdmin;
+use Sonata\AdminBundle\Templating\MutableTemplateRegistry;
+use Sonata\AdminBundle\Templating\MutableTemplateRegistryAwareInterface;
+use Sonata\AdminBundle\Templating\TemplateRegistryAwareInterface;
+use Sonata\AdminBundle\Templating\TemplateRegistryInterface;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\Definition;
+use Symfony\Component\DependencyInjection\Reference;
+
+/**
+ * @author Wojciech Błoszyk <wbloszyk@gmail.com>
+ */
+final class TemplateRegistryCompilerPass implements CompilerPassInterface
+{
+    const TEMPLATE_REGISTRY_TAG = 'sonata.admin.template_registry';
+    const MUTABLE_TEMPLATE_REGISTRY_TAG = 'sonata.admin.template_registry';
+    const TEMPLATE_REGISTRY_AWARE_TAG = 'sonata.admin.template_registry_aware';
```

Comment:
Both tags sound very technical / related to the concrete implementation. Maybe just `sonata.admin.template`?

Reply:
They look like related to the concreate implementation but thye have diffrence functionality too. 
"Nonmutable" tags sets template by constructor argument.
"Mutable" tags set template by `setTemplate` method.
"Nonaware" tags are for mark template registry service.
"Aware" tags will auto generate template registry.

I will think about it.